client/commonplugins/secrets_plugin.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package commonplugins
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/helper"
+)
+
+const (
+	SECRETS_DIR = "secrets"
```

Comment:
We already have something called a Secrets Dir for tasks, so maybe name this SecretsPluginDir just to make it more obvious? It definitely breaks Go's "don't repeat yourself in names" style preference, but since related code uses the _other_ Secrets Dir all over the place it might be worth being repetitive. Or not, nbd.


Probably worth switching to idiomatic Go const style of SecretsDir (and likewise for the consts below).